### ISSUE TITLE
jmeld SwingWorker thread hangs

### ISSUE THREAD

**Issue by WolfgangFahl** [albfan/jmeld#55]
I am not sure about how to handle the EDT/SwingWorker Issues with JMeld. 
Currently I have a situation where a diff will only show under certain timing conditions. 
In the real application JMeld hangs on first use. When debugging things work if a few milliseconds time are given but I do not know where yet. So far I found the culprit by pausing the SwingWorker Thread in Eclipse. and the stack shows that line 94 in FileComparison.doInBackground is the culprit. 
 
```
       if (contentPanel == null) {
                 diffNode.diff();
       }
```

eventually leads to StatusBar setFont which in line 1893 of Component.java has:
 
```
    public void setFont(Font f) {
        Font oldFont, newFont;
        synchronized(getTreeLock()) {
```

which waits for ever. I would appreciate some usage guidance on how to avoid this.

**Comment by albfan** [comment]
Any real use case to try to reproduce? I can test on Linux or windows

**Comment by WolfgangFahl** [comment]
I'll debug and see whether I can get a test case from the results.
https://stackoverflow.com/a/1792944/1497139 shows what I am assuming might be the  issue.

**Comment by WolfgangFahl** [comment]
I like this better:
```
diff --git a/src/main/java/org/jmeld/ui/FileComparison.java b/src/main/java/org/jmeld/ui/FileComparison.java
index fbd6503..f96e804 100644
--- a/src/main/java/org/jmeld/ui/FileComparison.java
+++ b/src/main/java/org/jmeld/ui/FileComparison.java
@@ -91,7 +91,11 @@ public class FileComparison extends SwingWorker<String, Object> {
             contentId = "BufferDiffPanel:" + diffNode.getId();
             contentPanel = JMeldPanel.getAlreadyOpen(mainPanel.getTabbedPane(), contentId);
             if (contentPanel == null) {
-                diffNode.diff();
+              SwingUtilities.invokeLater(new Runnable() {
+                public void run() {
+                  diffNode.diff(); 
+                }
+              });
             }
         } catch (Exception ex) {
             ex.printStackTrace();
```

and in JmDiffNode.diff avoiding to throw an exception with:
```
 public void diff()
  {
    BufferDocumentIF documentLeft;
    BufferDocumentIF documentRight;
    Object[] left, right;
    StatusBar statusBar = StatusBar.getInstance();

    statusBar.start();
    try {
      documentLeft = null;
      documentRight = null;

      if (nodeLeft != null) {
        documentLeft = nodeLeft.getDocument();
        statusBar.setState("Reading left : %s",
            nodeLeft.getName());
        if (documentLeft != null) {
          documentLeft.read();
        }
      }

      if (nodeRight != null) {
        documentRight = nodeRight.getDocument();
        statusBar.setState("Reading right: %s",
            nodeRight.getName());
        if (documentRight != null) {
          documentRight.read();
        }
      }

      statusBar.setState("Calculating differences");
      diff = new JMDiff();
      left = documentLeft == null ? null : documentLeft.getLines();
      right = documentRight == null ? null : documentRight.getLines();

      revision = diff.diff(left, right, ignore);
      statusBar.setState("Ready calculating differences");
    } catch (Exception ex) {
      statusBar.setState(String.format("Exception: %s -%s"+ex.getClass().getName(),ex.getMessage()));
    }
    StatusBar.getInstance().stop();
  }
```

**Comment by albfan** [comment]
Fixed on e5726a16e684ae76d053eebd2d51330de77268f3
